src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -45,6 +45,12 @@ object ScalafixPlugin extends AutoPlugin {
         "Optional list of custom rules to install from Maven Central. " +
           "This setting is read from the global scope so it only needs to be defined once in the build."
       )
+    val scalafixScalaVersion: SettingKey[String] =
+      settingKey[String](
+        "The full scala version used for the scalafix classpath. Rules must be compiled against that binary version." +
+          s"Defaults to the latest supported 2.12 version (${BuildInfo.scala212}). " +
+          "This setting is read from the global scope so it only needs to be defined once in the build."
```

Comment:
We could lift this constraint if it's too restrictive. I added it to avoid messing with the classloader structure right away, as (1) in https://github.com/scalacenter/sbt-scalafix/blob/v0.9.16/src/main/scala/scalafix/internal/sbt/ScalafixInterface.scala#L105-L120 could no longer be shared across subprojects if we were to allow per-project scala version for scalafix invocations, which would require some basic caching logic if we don't want performance to regress because of JIT warm-ups.